File: transpiler/javatests/com/google/j2cl/integration/java/whilestatement/Main.java
/*
 * Copyright 2015 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package whilestatement;

import static com.google.j2cl.integration.testing.Asserts.assertTrue;

/**
 * Test instanceof class type.
 */
public class Main {
  public static void main(String... args) {
    int count = 0;

    while (count < 5) {
      count++;
    }

    assertTrue(count == 5);

    while (count < 10) count++;
    assertTrue(count == 10);

    while (count++ < 14) {}
    assertTrue(count == 15);
  }
}

File: transpiler/javatests/com/google/j2cl/integration/kotlin/whilestatement/Main.kt
/*
 * Copyright 2022 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package whilestatement

import com.google.j2cl.integration.testing.Asserts.assertTrue

/** Test while loop. */
fun main(vararg unused: String) {
  var count = 0

  while (count < 5) {
    count++
  }
  assertTrue(5 == count)

  while (count < 10) count++
  assertTrue(10 == count)

  while (count++ < 14) {}
  assertTrue(15 == count)
}
